akka-javasdk/src/main/scala/akka/javasdk/impl/telemetry/Telemetry.scala

```diff
@@ -117,14 +116,13 @@ private[akka] object Telemetry {
 @InternalApi
 private[akka] object TraceInstrumentation {
   // Trick to extract trace parent from a single metadata entry and using the W3C decoding from OTEL
-  private val metadataEntryTraceParentGetter = new TextMapGetter[SpiMetadataEntry]() {
+  private val spiMetadataGetter = new TextMapGetter[SpiMetadata]() {
 
-    override def get(carrier: SpiMetadataEntry, key: String): String =
-      if (key == Telemetry.TRACE_PARENT_KEY) carrier.value
-      else null
+    override def get(carrier: SpiMetadata, key: String): String =
+      carrier.entries.find(_.key == key).map(_.value).orNull
 
-    override def keys(carrier: SpiMetadataEntry): lang.Iterable[String] =
-      Collections.singleton(Telemetry.TRACE_PARENT_KEY)
```

Comment:
Can we delete `TRACE_PARENT_KEY` now?

Reply:
Still needs some refactoring. Good point. When doing so I thought that maybe wasn't worth if you're move much of the tracing to the runtime. Shall I wait, or refactor anyway? WDYT?

Reply:
We'll still need to have context propagation on the SDK side, so probably useful to clean up.

For runtime <-> SDK propagation, I think this should just be `Context` objects passed directly. If we have SDK <-> external, without runtime control, then it should be full context propagation.